// default package
// Generated Sep 26, 2014 4:14:15 PM by Hibernate Tools 3.1.0.beta4

import java.util.List;
import javax.naming.InitialContext;
import org.apache.commons.logging.Log;
import org.apache.commons.logging.LogFactory;
import org.hibernate.LockMode;
import org.hibernate.SessionFactory;
import org.hibernate.criterion.Example;



/**
 * Home object for domain model class ClassTeacher.
 * @see .ClassTeacher
 * @author Hibernate Tools
 */
public class ClassTeacherHome {

    private static final Log log = LogFactory.getLog(ClassTeacherHome.class);

    private final SessionFactory sessionFactory = getSessionFactory();
    
    protected SessionFactory getSessionFactory() {
        try {
            return (SessionFactory) new InitialContext().lookup("SessionFactory");
        }
        catch (Exception e) {
            log.error("Could not locate SessionFactory in JNDI", e);
            throw new IllegalStateException("Could not locate SessionFactory in JNDI");
        }
    }
    
    public void persist(ClassTeacher transientInstance) {
        log.debug("persisting ClassTeacher instance");
        try {
            sessionFactory.getCurrentSession().persist(transientInstance);
            log.debug("persist successful");
        }
        catch (RuntimeException re) {
            log.error("persist failed", re);
            throw re;
        }
    }
    
    public void attachDirty(ClassTeacher instance) {
        log.debug("attaching dirty ClassTeacher instance");
        try {
            sessionFactory.getCurrentSession().saveOrUpdate(instance);
            log.debug("attach successful");
        }
        catch (RuntimeException re) {
            log.error("attach failed", re);
            throw re;
        }
    }
    
    public void attachClean(ClassTeacher instance) {
        log.debug("attaching clean ClassTeacher instance");
        try {
            sessionFactory.getCurrentSession().lock(instance, LockMode.NONE);
            log.debug("attach successful");
        }
        catch (RuntimeException re) {
            log.error("attach failed", re);
            throw re;
        }
    }
    
    public void delete(ClassTeacher persistentInstance) {
        log.debug("deleting ClassTeacher instance");
        try {
            sessionFactory.getCurrentSession().delete(persistentInstance);
            log.debug("delete successful");
        }
        catch (RuntimeException re) {
            log.error("delete failed", re);
            throw re;
        }
    }
    
    public ClassTeacher merge(ClassTeacher detachedInstance) {
        log.debug("merging ClassTeacher instance");
        try {
            ClassTeacher result = (ClassTeacher) sessionFactory.getCurrentSession()
                    .merge(detachedInstance);
            log.debug("merge successful");
            return result;
        }
        catch (RuntimeException re) {
            log.error("merge failed", re);
            throw re;
        }
    }
    
    public ClassTeacher findById( ClassTeacherId id) {
        log.debug("getting ClassTeacher instance with id: " + id);
        try {
            ClassTeacher instance = (ClassTeacher) sessionFactory.getCurrentSession()
                    .get("ClassTeacher", id);
            if (instance==null) {
                log.debug("get successful, no instance found");
            }
            else {
                log.debug("get successful, instance found");
            }
            return instance;
        }
        catch (RuntimeException re) {
            log.error("get failed", re);
            throw re;
        }
    }
    
    
    public List findByExample(ClassTeacher instance) {
        log.debug("finding ClassTeacher instance by example");
        try {
            List results = sessionFactory.getCurrentSession()
                    .createCriteria("ClassTeacher")
                    .add(Example.create(instance))
            .list();
            log.debug("find by example successful, result size: " + results.size());
            return results;
        }
        catch (RuntimeException re) {
            log.error("find by example failed", re);
            throw re;
        }
    } 

}